libdnf/sack/query.cpp

```diff
@@ -1757,7 +1783,8 @@ Query::Impl::filterLatest(const Filter & f, Map *m)
             Id p = samename.elements[i];
             considered = pool->solvables + p;
             if (!highest || highest->name != considered->name ||
-                ((keyname == HY_PKG_LATEST_PER_ARCH) && highest->arch != considered->arch)) {
+                ((keyname == HY_PKG_LATEST_PER_ARCH) && highest->arch != considered->arch) ||
+                ((keyname == HY_PKG_LATEST_PER_ARCH_BY_PRIORITY) && highest->arch != considered->arch)) {
```

Comment:
What about to make it more simple
```
((keyname == HY_PKG_LATEST_PER_ARCH || keyname == HY_PKG_LATEST_PER_ARCH_BY_PRIORITY) && highest->arch != considered->arch) {
```
or
```
bool latest_per_arch_keyname = keyname == HY_PKG_LATEST_PER_ARCH || keyname == HY_PKG_LATEST_PER_ARCH_BY_PRIORITY;
for (i = 0; i < samename.count; ++i) {
            Id p = samename.elements[i];
            considered = pool->solvables + p;
            if (!highest || highest->name != considered->name ||
                (latest_per_arch_keyname && highest->arch != considered->arch)) {
```

Reply:
I am really sorry but I have a bad feeling here. I believe that it will not work correctly when a single version of a package in two repositories with different priority. I believe that both packages will be returned by the filter. I will check it tomorrow.

Reply:
Don't worry, I won't merge it. I haven't even removed the "blocked" label yet, because I still have to test it for myself.

Reply:
Should be fixed now.